packages/ui/src/server/controllers/vcs.ts

```diff
@@ -50,24 +51,20 @@ const sendApiError = (res: Response, error: unknown) => {
     sendError(res, message, status || 500);
 };
 
-export const getVcsConfig: AppRouteHandler = (req, res) => {
-    const cookies = req.cookies;
+export const getVcsTokensAvailability: AppRouteHandler = (req: Request, res: Response) => {
+    const {cookies} = req;
     try {
-        const config = VcsFactory.getVcsSettings(req);
-        if (!config.length) throw new ErrorWithCode(404, ERROR_MESSAGE.VCS_CONFIG_NOT_FOUND);
-
-        const response = config.map<VcsConfig>((item) => ({
-            ...item,
-            hasToken: cookieNameByVcsId(item.id) in cookies,
-        }));
-        res.status(200).json(response);
+        const vcsTokenKeys = Object.keys(cookies).filter((name) =>
```

Comment:
It is better to avoid sending of names of cookies with tokens to browser. 
It is enough to send id of such VCS.